apollo-federation/src/merger/merger.rs

```diff
@@ -1208,8 +1231,104 @@ impl Merger {
         Ok(source_type.clone())
     }
 
-    pub(in crate::merger) fn merge_description<T>(&mut self, _sources: &Sources<T>, _dest: &T) {
-        todo!("Implement merge_description")
+    pub(in crate::merger) fn merge_description<T>(&mut self, sources: &Sources<T>, dest: &T)
+    where
+        T: HasDescription + std::fmt::Display,
+    {
+        let mut descriptions: CountMap<String, usize> = CountMap::new();
+
+        for (idx, source) in sources {
+            // Skip if source has no description
+            let Some(source_desc) = source
+                .as_ref()
+                .and_then(|s| s.description(self.subgraphs[*idx].schema()))
+            else {
+                continue;
+            };
+
+            descriptions.insert_or_increment(source_desc.trim().to_string());
+        }
+        descriptions.remove(&String::new());
+
+        if !descriptions.is_empty() {
+            // we don't want to raise a hint if a description is ""
+            if descriptions.len() == 1 {
+                if let Some((description, _)) = iter_into_single_item(descriptions.iter()) {
```

Comment:
This condition can replace `if descriptions.len() == 1`, since the two conditions are identical.